### PR TITLE
Closes #67 & #72 - Updates to ADVS template and vignette #78

### LINKED ISSUE #67: Feedback from testing: Small updates to ADVS Template and Vignette
## Background Information

The updates below were suggested from package testing.

### Template
- [x] Include in the label line a mention of "for Metabolic Trials" or something to that effect so that it clearly distinguishes from the admiral core ADVS template
- [x] Add `ASEQ` derivation
- [x] Add derivations for `BASECAT1` and `BASECA1N` to match vignette.
- [x] Ensure there is only ever at most one empty line at a time, in line with [tidyverse style guide](https://style.tidyverse.org/syntax.html#vertical-space).

### Vignette
- [x] Remove `advs <- advs %>% filter(!VSTESTCD == "BMI")` because it violates the data minimisation guidance (don't collect data which is not required for the analysis). Furthermore, the code is error-prone. If another parameter was derived before BMI, it would be removed as VSTESTCD is NA for this parameter.
- [x] Remove first paragraph of "Derive Categorization Variables (AVALCATx, BASECATx)" (`admiral does not currently have a generic function to aid in assigning AVALCATy/ AVALCAvN and BASECATy/ BASECAvN values. Below is a simple example of how these values may be assigned:`) because it contradicts the second one (`For deriving categorization variables (AVALCATx, BASECATx) admiral provides derive_vars_cat()  (see documentation of the function for details).`)
- [x] Always use text `AVALCATy` and `BASECATy` instead of `AVALCATx` and `BASECATx` to be in line with CDISC terminology.

### Template and Vignette
- [ ] Replace `constant_by_vars = exprs(USUBJID)`  with `get_admiral_option("subject_keys")`
- [x] Consider using `derive_var_base()` (or `derive_vars_merged()`) for deriving `BASECATy` and `BASECAyN`. This avoids defining the categories twice.
- [x] Renaming: `CRITyFN` instead of `CRITyFLN`.

### LINKED ISSUE #72: Feedback from testing: Add examples of using ratio functions to ADVS template and vignette
### Background Information

We forgot to showcase our ratio functions in the ADVS template and vignette.

### Definition of Done

- [ ] ADVS vignette updated with new section showcasing use of ratio functions
- [ ] ADVS template updated with code showcasing use of ratio functions (should match code in vignette)

### PR DESCRIPTION
**Pull request by AndersAskeland**
## Closes
Closes #67 
Closes #72 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by AndersAskeland** [comment]
@pharmaverse/admiralmetabolic If anyone has time, it would be great if someone could give this a quick review :)

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -272,25 +318,20 @@ dataset_vignette(
...
-  derive_vars_cat(
-    definition = basecat_lookup,
-    by_vars = exprs(PARAMCD)
+  derive_var_base(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+    source_var = AVALCAT1,
+    new_var = BASECAT1
+  ) |>
+  derive_var_base(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
```
Comment (yurovska) [inline vignettes/advs.Rmd:332]:
```suggestion
    by_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD),
```

File: vignettes/advs.Rmd [hunk 2]
```diff
@@ -212,11 +231,42 @@ advs <- derive_var_base(
...
+[`{admiralmetabolic}`](https://pharmaverse.github.io/admiralmetabolic/reference/derive_param_waisthip.html)
+provides several functions to quickly calculate various anthropometric
+ratios. For instance, the function
+[`derive_param_waisthip()`](https://pharmaverse.github.io/admiralmetabolic/reference/derive_param_waisthip.html)
+can be used to derive the waist-hip ratio.
+
+```{r eval=TRUE}
+advs <- advs %>%
+  derive_param_waisthip(
+    by_vars = exprs(USUBJID, VISIT, VISITNUM),
```
Comment (yurovska) [inline vignettes/advs.Rmd:250]:
```suggestion
    by_vars = exprs(!!!get_admiral_option("subject_keys"), VISIT, VISITNUM),
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R [hunk 3]
```diff
@@ -135,6 +129,20 @@ advs <- advs %>%
     constant_by_vars = get_admiral_option("subject_keys")
   )
 
+# Derive waist-hip ratio
+advs <- advs %>%
+  derive_param_waisthip(
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:134]: My rationale was that they are very similar in functionality and we probably don't want every admiralmetabolic derivation exported shown here. I imagine, if we at some point have a lot of parameter derivation functions, this could become messy.

File: inst/templates/ad_advs.R [hunk 4]
```diff
@@ -115,14 +110,13 @@ advs <- advs %>%
...
-# Derive domain specific variables ----
+# Derive domain specific parameters ----
 # See the "Derive Additional Parameters" vignette section for more information:
 # (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
 
-# Derive BMI
+# Derive BMI (remove BMI from source, and re-derive it)
 advs <- advs %>%
-  filter(VSTESTCD != "BMI") %>%
+  filter(VSTESTCD != "BMI" | is.na(VSTESTCD)) %>%
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:119]: I disagree here. While %in% is absolutely more concise, I think we should be more explicit as it is a template people might use.

[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:119]: I disagree here. While %in% is absolutely more concise, I think we should be more explicit as it is a template people might use.

File: inst/templates/ad_advs.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (yurovska) [reply inst/templates/ad_advs.R:134]: Makes sense. Resolved.

File: inst/templates/ad_advs.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:119]: I disagree here. While %in% is absolutely more concise, I think we should be more explicit as it is a template people might use.

[the remark this answers is not in the document]
Comment (yurovska) [reply inst/templates/ad_advs.R:119]: You're right. I double-checked with [admiral's Programming Concepts and Conventions](https://pharmaverse.github.io/admiral/articles/concepts_conventions.html#missing) and `!=` + 'is.na' is indeed mentioned there, so let's be consistent.

File: inst/templates/ad_advs.R [hunk 5]
```diff
@@ -256,6 +273,18 @@ advs <- advs %>%
...
+# Assign ASEQ ----
+# See the "Assign ASEQ" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#aseq)
+
+# Calculate ASEQ
+advs <- advs %>%
+  derive_var_obs_number(
+    new_var = ASEQ,
+    by_vars = get_admiral_option("subject_keys"),
+    order = exprs(PARAMCD, ADT, AVISITN, VISITNUM, ATPTN),
```
Comment (yurovska) [inline inst/templates/ad_advs.R:285]:
```suggestion
    order = exprs(PARAMN, ADT, AVISITN, VISITNUM, ATPTN),
```

File: vignettes/advs.Rmd [hunk 6]
```diff
@@ -272,25 +318,20 @@ dataset_vignette(
 )
 ```
 
-In a similar way, we will create `BASECATy`/ `BASECAvN` variables.
+Now we can use `derive_var_base` to derive the `BASECATy`/ `BASECAvN`
```
Comment (yurovska) [inline vignettes/advs.Rmd:321]:
```suggestion
Now we can use `derive_var_base` to derive the `BASECATy`/ `BASECAyN`
```

**Review by yurovska — CHANGES_REQUESTED** [review]
A couple more minors from my end. Otherwise, LTGM. Thanks for working on that!

**Comment by AndersAskeland** [comment]
Thanks for looking it over @yurovska ! 👍 

I will re-request review. If you approve, I can merge it.

**Review by yurovska — APPROVED** [review]
